src/Apps/ArtAdvisor/07-Curated-Discovery/weaviate-db.ts

```diff
@@ -0,0 +1,80 @@
+import { DiscoveryMarketingCollections } from "Apps/ArtAdvisor/07-Curated-Discovery/Components/Result/MarketingCollectionsRail"
+import _ from "lodash"
+import weaviate, { WeaviateClient } from "weaviate-ts-client"
+
+const DEFAULT_MARKETING_COLLECTIONS_CLASS = "DiscoveryMarketingCollections"
+
+export class WeaviateDB {
+  private client: WeaviateClient
+  private marketingCollectionClass: string
+
+  constructor(
+    options: {
+      /** Weaviate host url */
+      url?: string
+      /** Name of marketingCollection class to use. */
+      marketingCollectionClass?: string
+    } = {}
+  ) {
+    const url = options.url || process.env.WEAVIATE_URL
+    if (!url) throw new Error("Please provide url or set WEAVIATE_URL")
+
+    this.client = weaviate.client({ host: url })
+    this.marketingCollectionClass =
+      options.marketingCollectionClass || DEFAULT_MARKETING_COLLECTIONS_CLASS
+  }
+
+  /**
+   * Fetch marketing collections by semantic search on an array of concepts.
+   */
+  async getNearMarketingCollections({
+    concepts,
+    budget,
+    limit = 10,
+  }: {
+    /** List of concepts for semantic search */
+    concepts: string[]
+    /** Budget on which to filter collections */
+    budget?: number
```

Comment:
We need to decide what strategy to use when filtering. We have the `priceGuidance` [field](https://github.com/artsy/gravity/blob/006bb5e97875f0b13a06592121302349f158c5b3/config/initializers/rails_admin.rb#L712) available to us in `weaviate`, so we can use that price in our filtering strategy.